sklearn/svm/tests/test_bounds.py

```diff
@@ -74,3 +76,41 @@ def test_ill_posed_min_c():
 def test_unsupported_loss():
     with pytest.raises(ValueError):
         l1_min_c(dense_X, Y1, loss='l1')
+
+
+# TODO test deterministic results on differing systems
+# def test_set_seed():
+#     def _test(seed, val):
+#         # TODO currently causes 99% coverage in test_bounds.py
+#         if seed is not None:
+#             set_seed_wrap(seed)
+#         x = bounded_rand_int_wrap(100)
+#         assert(x == val), 'Expected {} but got {} instead'.format(val, x)
+
+#     # TODO should be default seed for std::mt19937,
+#     # but different results on jupyter lab
+#     # _test(None, 81)
+#     _test(5489, 81)  # default seed for std::mt19937
+#     _test(0, 54)
+#     _test(4294967295, 9)  # max unsigned int size
+
+
+_MAX_INT = 2147483647
+
+
+@pytest.mark.parametrize("orig_range", [_MAX_INT, 100])
+def test_bounded_rand_int(orig_range):
+    n_pts = 1000
+    n_iter = 100
+    ks_pvals = []
+    uniform_dist = stats.uniform(loc=0, scale=orig_range)
+    # avg over multiple tests to make chance of outlier result negligible
+    for _ in range(n_iter):
+        # Deterministic random sampling
+        sample = [bounded_rand_int_wrap(orig_range) for _ in range(n_pts)]
+        res = stats.kstest(sample, uniform_dist.cdf)
+        ks_pvals.append(res.pvalue)
+    avg_pval = np.mean(ks_pvals)
```

Comment:
We can instead use a quantile. In my tests, using the lower 10% quantile seems to pass:

 `min_10pct_pval = np.quantile(ks_pvals, q=0.1)`

Reply:
Good idea, any reason 10% quantile in particular?

Reply:
At first I was expecting the minimum to be ok (as you first tried). However this did not work.
So I tried to find the lowest possible quantile that works :)

I finally found the right scientific explanation in https://stats.stackexchange.com/questions/298350/uniform-distribution-testing

> That is how all hypothesis tests work. Under the null hypothesis, p is uniformly distributed.

We should probably include this as a comment, as it is non-trivial for non-statistics experts.

So maybe "the" best way to do the final test (in addition to the current assert on quantile, which can not hurt) is to do a K-S uniformity test on the p-values themselves. They should be uniform on [0, 1] if null hypothesis is not rejected.

I felt I learned something today :)